core/domain/collection_services.py

```diff
@@ -0,0 +1,872 @@
+            collection_summary.collection_model_last_updated),
+        collection_model_created_on=(
+            collection_summary.collection_model_created_on)
+    )
+
+    collection_summary_model.put()
+
+
+def delete_collection_summary(collection_id, force_deletion=False):
+    """Delete an collection summary model."""
+
+    collection_models.CollectionSummaryModel.get(collection_id).delete()
+
+
+def save_new_collection_from_yaml(committer_id, yaml_content, collection_id):
+    collection = collection_domain.Collection.from_yaml(
+        collection_id, yaml_content)
+    commit_message = (
+        'New collection created from YAML file with title \'%s\'.'
+        % collection.title)
+
+    _create_collection(committer_id, collection, commit_message, [{
+        'cmd': CMD_CREATE_NEW,
+        'title': collection.title,
+        'category': collection.category,
+    }])
+
+    return collection
+
+
+def delete_demo(collection_id):
+    """Deletes a single demo collection."""
+    if not (0 <= int(collection_id) < len(feconf.DEMO_COLLECTIONS)):
+        raise Exception('Invalid demo collection id %s' % collection_id)
+
+    collection = get_collection_by_id(collection_id, strict=False)
+    if not collection:
+        logging.info('Collection with id %s was not deleted, because it '
+                     'does not exist.' % collection_id)
+    else:
+        delete_collection(
+            feconf.SYSTEM_COMMITTER_ID, collection_id, force_deletion=True)
+
+
+def load_demo(collection_id):
+    """Loads a demo collection.
+
+    The resulting collection will have version 2 (one for its initial
+    creation and one for its subsequent modification.)
+    """
+    delete_demo(collection_id)
+
+    if not collection_domain.Collection.is_demo_collection_id(collection_id):
+        raise Exception('Invalid demo collection id %s' % collection_id)
+
+    demo_filepath = os.path.join(
+        feconf.SAMPLE_COLLECTIONS_DIR,
+        feconf.DEMO_COLLECTIONS[int(collection_id)])
+
+    if demo_filepath.endswith('yaml'):
+        yaml_content = utils.get_file_contents(demo_filepath)
+    else:
+        raise Exception('Unrecognized file path: %s' % demo_filepath)
+
+    collection = save_new_collection_from_yaml(
+        feconf.SYSTEM_COMMITTER_ID, yaml_content, collection_id)
+
+    rights_manager.publish_collection(
+        feconf.SYSTEM_COMMITTER_ID, collection_id)
+
+    index_collections_given_ids([collection_id])
+
+    # Now, load all of the demo explorations that are part of the collection.
+    for collection_node in collection.nodes:
+        exp_id = collection_node.exploration_id
+        # Only load the demo exploration if it is not yet loaded.
+        if exp_services.get_exploration_by_id(exp_id, strict=False) is None:
+            exp_services.load_demo(exp_id)
+
+    logging.info('Collection with id %s was loaded.' % collection_id)
+
+
+# TODO(bhenning): Cleanup search logic and abstract it between explorations and

```

Comment:
As we discussed offline, we want to avoid tightly coupling search functionality between explorations and collections so this is not going to be done as part of this branch.